core/src/main/java/org/infinispan/configuration/cache/Configurations.java

```diff
@@ -19,11 +19,19 @@ private Configurations() {
    }
 
    public static boolean isOnePhaseCommit(Configuration cfg) {
+      // Otherwise pessimistic transactions will be one phase commit
+      if (cfg.memory().size() > 0 && cfg.memory().evictionType().isExceptionBased()) {
+         return false;
+      }
       return !cfg.clustering().cacheMode().isSynchronous() ||
             cfg.transaction().lockingMode() == LockingMode.PESSIMISTIC;
    }
 
    public static boolean isOnePhaseTotalOrderCommit(Configuration cfg) {
+      // Even total order needs 2 phase commit with this - isn't even worth it then?
```

Comment:
Good point. Maybe we should spit out a warning to the user? Something like - "TotalOrder optimization will be disabled due to Blocking Eviction".

Reply:
I can add that to the validate method on the ConfigurationBuilder maybe. I guess it should be an INFO message or do you think WARN?